test/e2e/extended_suite_test.go

```diff
@@ -5,37 +5,39 @@ package e2e
 import (
 	"flag"
 	"fmt"
-	"os"
-	"strings"
 	"testing"
 
+	"github.com/kelseyhightower/envconfig"
 	. "github.com/onsi/ginkgo"
 	"github.com/onsi/ginkgo/config"
 	. "github.com/onsi/gomega"
 )
 
 var (
 	gitCommit   = "unknown"
-	kubeconfig  = flag.String("kubeconfig", "../../_data/_out/admin.kubeconfig", "Location of the kubeconfig")
+	testConfig  Config
 	artifactDir = flag.String("artifact-dir", "", "Directory to place artifacts when a test fails")
 )
 
 var _ = BeforeSuite(func() {
-	c = newTestClient(*kubeconfig, *artifactDir)
+	err := envconfig.Process("", &testConfig)
+	if err != nil {
+		panic(err)
+	}
+	testConfig.ArtifactDir = *artifactDir
 
-	focus := []byte(config.GinkgoConfig.FocusString)
-	if strings.Contains(string(focus), "\\[CustomerAdmin\\]") {
-		_, err := os.Stat("../../_data/_out/customer-cluster-admin.kubeconfig")
-		if err != nil {
-			panic(err)
-		}
-		_, err = os.Stat("../../_data/_out/customer-cluster-reader.kubeconfig")
-		if err != nil {
-			panic(err)
-		}
-		creader = newTestClient("../../_data/_out/customer-cluster-reader.kubeconfig", *artifactDir)
-		cadmin = newTestClient("../../_data/_out/customer-cluster-admin.kubeconfig", *artifactDir)
+	suiteSummary := config.GinkgoConfig.FocusString
+	if inFocus(suiteSummary, "CustomerAdmin") {
+		c = newTestClient(testConfig.EndUserKubeConfig, testConfig.ArtifactDir)
+		creader = newTestClient(testConfig.CustomerClusterReaderKubeConfig, testConfig.ArtifactDir)
+		cadmin = newTestClient(testConfig.CustomerClusterAdminKubeConfig, testConfig.ArtifactDir)
+		return
+	}
+	if inFocus(suiteSummary, "AzureClusterReader") {
```

Comment:
I would prefer we still honor KUBECONFIG for these and not have separate clients/kubeconfig entries.

Reply:
This is only possible if we have only one client per test category. So we might want to break the customer admin tests into smaller categories.

Reply:
The customer admin tests are a special case I think and we will not be able to get rid of the end user kubeconfig from them (unless we use auth delegation which is not ideal). The rest of the tests (end user, azure reader) do not need a second kubeconfig so it makes sense for them to honor KUBECONFIG (we already run all these suites separately in hack/e2e.sh).

Reply:
@jim-minter @mjudeikis any feelings here?

Reply:
I think this is needed, but this looks very complicated. I feel we need to have smaller chunks where we can and honour `KUBECONFIG` and keep `KUBECONFIG_` to minimal usage. Maybe we need to rethink little bit our test structure?

Reply:
Not sure it's necessary to do it here and I cannot see how we get around the fact that we need multiple kubeconfigs in a single test run. FWIW, our approach is better than what has been used in the origin e2e tests historically (a single bash script that jumps around contexts via oc login).

Reply:
@kargakis @mjudeikis what do you think about the redesign?

Reply:
I like this more. maybe just rename kubeconfig from `customer-cluster-admin.kubeconfig` to something saying that now it was all context inside. as its not just cluster-admin now,